ADotNet/Models/Pipelines/GithubPipelines/DotNets/TagJob.cs

```diff
@@ -0,0 +1,28 @@
+﻿// ---------------------------------------------------------------------------
+// Copyright (c) Hassan Habib & Shri Humrudha Jagathisun All rights reserved.
+// Licensed under the MIT License.
+// See License.txt in the project root for license information.
+// ---------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using ADotNet.Models.Pipelines.GithubPipelines.DotNets.Tasks;
+using YamlDotNet.Serialization;
+
+namespace ADotNet.Models.Pipelines.GithubPipelines.DotNets
+{
+    public partial class TagJob
+    {
+        [YamlMember(Alias = "if")]
+        public string TagTrait = "github.event_name == 'pull_request' && github.event.action == 'closed'" +
```

Comment:
This did not work for me.  Can you provide some screenshots to show this is working only at the times you need it.

Reply:
we should also not hardcode this.  consumer should be able to build this section out themselves.

Reply:
Will try to find some time this weekend to get a test repo setup as suggested.